scripts/quests/windurst/Carbuncle_Debacle.lua

```diff
@@ -0,0 +1,186 @@
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/status')
+require('scripts/globals/interaction/quest')
+require('scripts/globals/titles')
+-----------------------------------------------
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.CARBUNCLE_DEBACLE)
+--------------------------------------------------------------------------------------
+quest.reward =
+{
+    item            =   xi.items.EVOKERS_HORN,
+    fame            =   60,
+    fameArea        =   xi.quest.fame_area.WINDURST,
+}
+
+quest.sections =
+{
+    --Section: Quest Available
+
+    {
+
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getMainLvl() >= xi.settings.main.AF3_QUEST_LEVEL and
+                player:getMainJob() == xi.job.SMN and
+                not quest:getMustZone(player)
+        end,
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['_6n2'] = quest:progressEvent(415),
+
+                onEventFinish =
+                {
+                    [415] = function(player, csid, option, npc)
+                            quest:begin(player)
+                            quest:setVar(player, 'Prog', 1)
+                    end,
+                },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        --Section: Quest Accepted
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['Koru-Moru'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        return quest:event(416)
+                    elseif questProgress == 4 then
+                        return quest:progressEvent(417)
+                    elseif questProgress == 5 then
+                        return quest:progressEvent(418)
+                    elseif questProgress == 7 then
+                        return quest:progressEvent(419)
+                    elseif player:getQuestStatus(xi.quest.log_id.WINDURST, xi.quest.id.windurst.CARBUNCLE_DEBACLE) == QUEST_COMPLETED then
+                        return player:event(420)
+                    end
+                end,
+            },
+                onEventFinish =
+                {
+                    [416] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 2)
+                    end,
+
+                    [417] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 5)
+                        npcUtil.giveKeyItem(player, xi.ki.DAZE_BREAKER_CHARM)
+                    end,
+
+                    [419] = function(player, csid, option, npc)
+                        quest:complete(player)
+                        player:addTitle(xi.title.PARAGON_OF_SUMMONER_EXCELLENCE)
```

Comment:
Best to put the complete into an if statement as it rewards an item. This prevents players from getting the title without actually completing the quest.

```suggestion
                        if quest:complete(player) then
                            player:addTitle(xi.title.PARAGON_OF_SUMMONER_EXCELLENCE)
                        end
```

Reply:
Moreover, we can actually just put the title into the quest.reward table

Reply:
title    = xi.title.PARAGON_OF_SUMMONER_EXCELLENCE,